src/chains/evm/hub/utils/message.ts

```diff
@@ -114,21 +119,32 @@ export async function getHubReverseMessageExtraArgsAndAdapterFees(
 }> {
   const { action, data } = payload;
   const payloadData = decodeMessagePayloadData(action as ReversibleHubAction, data);
+  const folksTokenId = getFolksTokenIdFromPool(payloadData.poolId);
 
-  const returnAdapterId = extraArgsParams?.returnAdapterId ?? message.returnAdapterId;
+  const returnAdapterId =
+    extraArgsParams?.returnAdapterId ??
+    getRandomElement(
+      getSupportedMessageAdapters({
+        messageAdapterParamType: MessageAdapterParamsType.SendToken,
+        action: action as ReversibleHubAction,
+        network,
+        folksTokenId,
+        sourceFolksChainId: message.sourceChainId,
+      }).adapterIds,
```

Comment:
I don't think this is correct, this is the adapters for sending a token from spoke to hub

Reply:
yes, in fact the spoke is set as `sourceFolksChainId` so that the correct adapters are returned, the alternative is to set as default adapter that one used to send the message from spoke to hub, what do you think?

Reply:
Yes I think it's fine how it was. If the one used to send the message from spoke to hub is incorrect then they are able to use extraArgs still

Reply:
0c6ad1d0fc3d72c00109ec4d6a6d698e57de12da

Reply:
I meant if the `returnAdapterId` when sending the message from to hub is incorrect..., not to use the `adapterId`